nodejs/README.md

```diff
@@ -0,0 +1,208 @@
+  allows more folks to jump in.
+- Front end build tools make it easy to choose a syntax you want and polyfill
+  the missing features in the browser.
+
+
+## Frameworks
+
+Frameworks provide a common baseline for your application. ATO requires
+configurations that most frameworks provide out of the box, or make it very easy
+to enable.
+
+The most common framework at 18F is [Express][expressjs]. You don't have to
+choose Express, but you should use an existing framework with an active
+community for support.
+
+Here are some existing projects at 18F using Express:
+
+- [Federalist][github-federalist]
+- [OMB eRegs][github-omb-eregs]
+- [Analytics Reporter API][github-analytics-report]
+
+You can [find many
+more](https://github.com/search?q=org%3A18F+filename%3Apackage.json+express&type=Code&utf8=%E2%9C%93) internal and
+external projects as well.
+
+
+### Best practices
+
+
+#### Recommend
+
+- Use an existing framework with active community support.
+
+
+#### Suggest
+
+- Use [Express][expressjs].
+
+
+## Scaffolding your application
+
+Scaffolding tools can help take care of some of boiler plate of initializing
+a project. Some scaffolds are generator tools, others are kits intended to be
+forked. There are many available on Github. You might also choose one based on the
+front end frameworks you want to use and then add the server-side pieces
+yourself. You also don't have to use one at all.
+
+Here's a list of scaffolds we've had success with:
+
+- [express-generator](https://www.npmjs.com/package/express-generator)
+- [express-babel](https://github.com/vmasto/express-babel)
+
+
+### Best practices
+
+
+#### Consider
+
+- What front end frameworks do you plan to use? Maybe choose a generator based
+  on those frameworks instead.
+- Don't spend too much time finding the perfect generator. Once your application
+  has been scaffolded, you're not going to scaffold it again.
+
+
+## Authentication
+
+For Express, [Passport](http://www.passportjs.org/) has many authentication
+plugins including [OAuth2](https://www.npmjs.com/package/passport-oauth2) which
+can be used with cloud.gov authentication.
+
+
+### Best practices
+
+#### Recommend
+
+- Avoid writing your own authentication. Use an existing framework with an
+  active development community.
+
+
+#### Suggest
+
+- Use [Passport](http://www.passportjs.org/) for your Express application.
+
+
+## Security
+
+You'll want some basic web application hardening.
+
+[helmet](https://www.npmjs.com/package/helmet) includes several smaller
+middleware functions to set secure HTTP headers on your application.
+
+If you're using any kind of user sessions, use
+[cookie-session](https://www.npmjs.com/package/cookie-session) for secure,
+encrypted, signed session cookies.
+
+When used with session middleware, [csurf](https://www.npmjs.com/package/csurf)
+provides cross-site request forgery (CSRF) protection.
+
+For database queries, use a library or ORM like
```

Comment:
http://knexjs.org/ is a pretty great query builder library if you want to have a non-ORM example here.